news.js

```diff
@@ -0,0 +1,244 @@
+    },
+    {
+        image: "<div class='content__img'></div>",
+        heading: "Fox News",
+        date: "Posted on 29 June, 2019 //Category: Category One",
+        content: "Djokovic reveals ‘silly’ mental trick to help him deal with crowd at Wimbledon",
+        popup: `Djokovic revealed a mental trick, according to the Times. After winning the match, after nearly five hours, he told reporters that when the crowd chanted “Roger! Roger,” he said he heard “Novak.”“It sounds silly, but it is like that. I try to convince myself that it’s like that.” The sets were 7-6 (5), 1-6, 7-6 (4), 4-6 and 13-12 (3). The final lasted nearly five hours, longer than any other Wimbledon final. Djokovic has now won Wimbledon five times.`
+    },
+    {
+        image: "<div class='content__img'></div>",
+        heading: "CNN",
+        date: "Posted on 29 June, 2019 //Category: Category One",
+        content: "'Silverstone, I love you': Lewis Hamilton wins record sixth British Grand Prix",
+        popup: `Lewis Hamilton claimed a thrilling victory at Silverstone on Sunday to win a record-breaking sixth British Grand Prix. It looked as though teammate Valtteri Bottas would spoil the party for the 141,000 visiting fans, but a timely safety car intervention gifted Hamilton first place. Ferrari's Charles Leclerc claimed an impressive third place after an intense, wheel-to-wheel battle with Max Verstappen throughout the race.`
+    }
+
+];
+//Channels present in the dropdown list
+var channels = ["BBC", "Aaj Tak", "NBC", "NDTV", "Republic", "TV9", "India Today", "Times Now", "Fox News", "CNN"];
+//variable used to store the complete data of all news channels
+var fulldata = " ";
+
+//Function to display the selected news category
+function myFunction() {
+    var x = document.getElementById("sel-category").value;
+    for (let i = 0; i < data.length; i++) {
+        if (x === data[i].heading) {
+            let display = "<div class='content' id='content'>" +
+                "<div class='content__sub' id='content__display'>" +
+                data[i].image +
+                "<h3 class='content__modifier content__head'>" + data[i].heading + "</h3>" +
+                "<p class='content__modifier content__date'>" + data[i].date + "</p>" +
+                "<p class='content__modifier content__matter'>" + data[i].content + "</p> " +
+                "<a href='#' class='content__modifier btn btn--pink' id='myBtn' onclick='showpopup()'>Continue Reading</a>" +
+                "</div>" +
+                `<div id="myModal" class="modal">
```

Comment:
You dont need to construct the modal for every data. Create a blank modal. On click of read more pass the id/index of the data. then showPopup can use that id/index to get the data and display.